java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -577,17 +580,23 @@ public void visitNewClass(NewClassTree tree) {
       }
       scan(classBody);
       registerType(tree, anonymousClassType);
+      // } else if (callToInitWithDiamondOperator(tree.identifier()) && !resolution.type().isUnknown()) {
+      // registerType(tree, ((MethodJavaType) resolution.type()).resultType);
     } else {
       registerType(tree, getType(tree.identifier()));
     }
   }
 
-  private JavaSymbol resolveConstructorSymbol(IdentifierTree identifier, Type type, Resolve.Env methodEnv, List<JavaType> argTypes) {
-    Resolve.Resolution resolution = resolve.findMethod(methodEnv, (JavaType) type, "<init>", argTypes);
+  private static boolean callToInitWithDiamondOperator(TypeTree typeTree) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private method 'callToInitWithDiamondOperator' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)